### PR TITLE
Relax PostgreSQL connection checks

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Allow more locales, improve error messages and documentation.

## Readiness checklist

* [x] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [x] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [x] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: website/docs/understanding-ferretdb.md [hunk 1]
```diff
@@ -158,9 +158,15 @@ As with any database, before moving to production, please verify if it is suitab
...
-MongoDB collections are mapped to PostgreSQL tables,
-and MongoDB documents are mapped to rows with a single [JSONB](https://www.postgresql.org/docs/current/datatype-json.html) column.
-The physical layout might change as we work on improving compatibility and performance.
+
+PostgreSQL should be configured with `UTF8` encoding and one of the following locales:
+`POSIX`, `C`, `C.UTF8`, `en_US.UTF8`.
+
+MongoDB databases are mapped to PostgreSQL schemas in a single PostgreSQL database that should be created in advance.
+MongoDB collections are mapped to PostgreSQL tables.
+MongoDB documents are mapped to rows with a single [JSONB](https://www.postgresql.org/docs/current/datatype-json.html) column.
```
Comment (rumyantseva) [inline website/docs/understanding-ferretdb.md:167]: In these three sentences, we use "MongoDB" word, but in the code itself, we call these things like "FerretDB databases" and "FerretDB collections". I'm ok with either option though.

**Review by rumyantseva — APPROVED** [review]
LGTM, one minor question.

File: website/docs/understanding-ferretdb.md (hunk 1 above)
Comment (AlekSi) [reply website/docs/understanding-ferretdb.md:167]: Code and comments are for developers, but that documentation is for users

**Review by chilagrow — APPROVED** [review]

**Comment by AlekSi** [comment]
@Fashander PTAL

**Review by Fashander — APPROVED** [review]

**Pull request merged**